### PR TITLE
Use informers

### PR DESCRIPTION
**Pull request by rahanar**
This PR depends on PR #15.

This reimplements src-dst controller to use informers and workqueue. Will fix/add more tests and test on a cluster.

### CODE REVIEW

File: pkg/controller/srcdst_controller.go [hunk 1]
```diff
@@ -4,110 +4,154 @@ import (
...
 
-func (c *Controller) disableSrcDstIfEnabled(node *v1.Node) {
-	srcDstCheckEnabled := true
-	if node.Annotations != nil {
-		if _, ok := node.Annotations[SrcDstCheckDisabledAnnotation]; ok {
-			srcDstCheckEnabled = false
+func (c *Controller) disableSrcDstCheck(key string) error {
+	defer c.workqueue.Done(key)
+
+	return retry.RetryOnConflict(retry.DefaultBackoff, func() error {
```
Comment (ottoyiu) [inline pkg/controller/srcdst_controller.go:125]: is it possible to have a less greedy retry that does not calling 'c.modifySrcDstCheckAttribute' again if there's a conflict?

File: pkg/controller/srcdst_controller.go [hunk 2]
```diff
@@ -139,8 +182,38 @@ func GetInstanceIDFromProviderID(providerID string) (*string, error) {
...
+
+	if !exists {
+		return nil, fmt.Errorf("node object %s doesn't exist", key)
+	}
+
+	return nodeItem.(*v1.Node), nil
+}
+
+func (c *Controller) checkSrcDstAttributeEnabled(key string) (enabled bool, err error) {
+	defer c.workqueue.Done(key)
```
Comment (ottoyiu) [inline pkg/controller/srcdst_controller.go:205]: Having this method call 'c.workqueue.Done' is a bit hard to follow.  Perhaps it can be placed before :
https://github.com/ottoyiu/k8s-ec2-srcdst/pull/16/files#diff-bff40c211a4e0c5aed5c55ab3cdbda41R10

and have the failure cases on 'processNextWorkItem' return early instead.

File: pkg/controller/srcdst_controller.go [hunk 3]
```diff
@@ -4,110 +4,154 @@ import (
...
+	return
 }
 
-func (c *Controller) disableSrcDstIfEnabled(node *v1.Node) {
-	srcDstCheckEnabled := true
-	if node.Annotations != nil {
-		if _, ok := node.Annotations[SrcDstCheckDisabledAnnotation]; ok {
-			srcDstCheckEnabled = false
+func (c *Controller) disableSrcDstCheck(key string) error {
+	defer c.workqueue.Done(key)
```
Comment (ottoyiu) [inline pkg/controller/srcdst_controller.go:123]: likewise to my other comment regarding modifying the workqueue in this method.

**Review by ottoyiu — COMMENTED** [review]
Some minor comments but otherwise lgtm (apart from test coverage)

**Comment by gjtempleton** [comment]
Hey @ottoyiu really appreciate all your work on maintaining this project. Is there any chance of either this or #13 getting merged in the near future?

**Comment by rahanar** [comment]
hey @gjtempleton, I will work on merging this soon!
